src/services/auth/auth.ts

```diff
@@ -0,0 +1,57 @@
+import { AuthProviders } from '@prisma/client';
+
+import * as crypt from '@/lib/crypt';
+import { findByProvider } from '@/repositories/identity';
+import { findByEmail } from '@/repositories/user';
+import { Credentials } from '@/types/common';
+
+import AuthError from './error';
+import { createUserWithIdentity, mergeExitingUser } from '../user/user';
+
+interface Profile {
+  email?: string;
+  name?: string;
+}
+
+export async function authenticateLogin(credentials: Credentials) {
```

Comment:
The method name is hard to differentiate (especially when checking the tests). How about using the word `credential` as `Google` is used in the second method:

```suggestion
export async function authenticateCredentialLogin(credentials: Credentials) {
```